src/Plugin/Field/FieldFormatter/ErrandServicesFormatter.php

```diff
@@ -0,0 +1,142 @@
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    return [
+        'sort_order' => [],
+      ] + parent::defaultSettings();
+  }
+
+  /**
+   * Gets the channel types.
+   *
+   * @return \Drupal\helfi_tpr\Entity\ChannelTypeCollection
+   *   The channel type collection.
+   */
+  private function getChannelTypes() : ChannelTypeCollection {
+    return ChannelTypeCollection::createFromArray($this->getSetting('sort_order'));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    $form = parent::settingsForm($form, $form_state);
+
+    $form['sort_order'] = [
+      '#type' => 'table',
+      '#caption' => $this->t('Order'),
+      '#header' => [
+        $this->t('ID'),
+        $this->t('Label'),
+        $this->t('Weight'),
+      ],
+      '#tableselect' => FALSE,
+      '#tabledrag' => [
+        [
+          'action' => 'order',
+          'relationship' => 'sibling',
+          'group' => 'group-weight',
+        ],
+      ],
+    ];
+
+    foreach ($this->getChannelTypes() as $item) {
+      $form['sort_order'][$item->id] = [
+        '#attributes' => ['class' => ['draggable']],
+        '#weight' => $item->weight,
+        'id' => ['#plain_text' => $item->id],
+        'label' => ['#plain_text' => $item->label()],
+        'weight' => [
+          '#type' => 'weight',
+          '#title' => $this->t('Weight for @title', ['@title' => $item->label()]),
+          '#title_display' => 'invisible',
+          '#default_value' => $item->weight,
+          '#attributes' => ['class' => ['group-weight']],
+        ],
+      ];
+    }
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) : array {
+
+    if (!$items->getEntity() instanceof Service) {
+      throw new \InvalidArgumentException('The field can only be used with tpr_errand_service entities.');
+    }
+
+
+//    $elements = parent::viewElements($items, $langcode);
+    $services = $items->getEntity();
+    $channelTypes = $this->getChannelTypes();
+    $language = \Drupal::languageManager()->getCurrentLanguage()->getId();
+
+
+    $elements = [];
+    $errand_services = $items->referencedEntities();
+
+    foreach ($errand_services as $errand_service) {
+      foreach ($errand_service->getChannels() as $channel) {
+
+
+      if (in_array($channel->getType(), $elements)) {
+        continue;
+      }
+      $translatedChannel = $channel->getTranslation($language);
+
+      $elements[] = [
+        '#name' => $translatedChannel->type_string->value,
+        '#weight' => $channelTypes[$channel->getType()]->weight,
+      ];
+    }
+    }
+    if ($items->getEntity()->has_unit->value) {
+      $elements[] = [
+        '#name' => $this->t('Office'),
+        '#weight' => 999,
+      ];
+    }
+    uasort($elements, [SortArray::class, 'sortByWeightProperty']);
+
+    $list = [];
```

Comment:
you could use array_column directly in '#items' => , instead of having this extra foreach